Assets/__Scripts/MapEditor/UI/Extensions/ExtensionButtons.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public static class ExtensionButtons
+{
+    private static List<ExtensionButton> buttons = new List<ExtensionButton>();
+
+    public static ExtensionButton AddButton(Sprite icon, string tooltip, UnityAction onClick)
```

Comment:
Why supply these properties to this function rather than an ExtensionButton itself?
Seems like it would be more future proof to use a builder-esque pattern where you construct the button and then supply it so old plugins don't break if you add a new property for instance.

Reply:
Main idea was to keep it simple for the majority of use cases since those properties are the only 3 most plugins will touch. This way most of them can even just ignore the returned ExtensionButton since they'll never use it. Having another AddButton that just takes an ExtensionButton would be a good idea though.

Reply:
PR otherwise looks good, I'll wait for this to be resolved first before merging.